linker/Mono.Linker/Annotations.cs

```diff
@@ -56,6 +56,18 @@ public class AnnotationStore {
 		System.Xml.XmlWriter writer;
 		GZipStream zipStream;
 
+		public void Clear ()
```

Comment:
We have an experimental feature where we do multiple passes.  The way it's implemented is to construct a Pipeline that registers steps multiple times.  We need to clear the annotations between each pass.  I'm guessing the reasoning was that clearing the annotations was preferable over replace the Annotations instance stored on the LinkContext.

Here is what our GetStandardPipeline logic looks like to help show what we are doing.

```
        private static Pipeline GetStandardPipeline()
        {
            Pipeline p = new Pipeline();
            // TODO: Don't run repeatedly. Causes issues
            //for (int i=0; i!=3; i++)
            {
                p.AppendStep(new ClearAnnotationStoreStep());
                p.AppendStep(new LoadReferencesStep());
                p.AppendStep(new BlacklistImplicitlyFromFilesStep());
                p.AppendStep(new ResolveFromMonoBehaviours(LinkerOptions.Api));
                p.AppendStep(new ResolveFromPreserveAttribute());
                p.AppendStep(new PreserveIntPtrFields());

                // The upstream black list step will black list based on the .xml resources embedded in the Mono.Linker
                // executable.  We don't want this as it results in preserving a bunch of stuff that we don't care about
                //p.AppendStep(new BlacklistStep());

                p.AppendStep(new TypeMapStep());
                //p.AppendStep(new DevirtualizeStep());
                p.AppendStep(new UnityMarkStep());
                p.AppendStep(new UnitySweepStep());
                p.AppendStep(new CleanStep());
            }

            // Keep this outside the multi-pass loop.  It only needs to run once at the end,
            // running it multiple times results in duplicate types being added
            p.AppendStep(new AddUnresolvedStubsStep());

            p.AppendStep(new RegenerateGuidStep());
            p.AppendStep(new UnityOutputStep());
            return p;
        }
```

As you can see, we have the loop commented out right now.  I don't know what the issues were, it's on my list of things to revisit.

We can close this PR if you would prefer we wait until we have a need for it.